### PR TITLE
Task 6: Swin Transformer Alzheimer's Classification by Tarushi Gera (ID: 48242204)

### PR DESCRIPTION
**Pull request by Tarushi263**
## TASK 6

**Classify Alzheimer’s disease (normal and AD) of the ADNI brain data (see Appendix for link) using a visual or perceiver transformer set having a minimum accuracy of 0.8 on the test set.**
**Pull Request Comment:**

🚀 **Pull Request Summary:**

In this project, I have attempted to classify the input images by preprocessing them and passing them through a Swin Architecture Model which has been referenced. I have attempted to define all the hyperparameter tuning components in the code as well as the README.md file. The project also contains a config.yaml file which can be utilized to make custom configurations to run the project. This pull request appeals to receive a feedback and merge my version of Alzheimer's Disease Classification with the Swin Transformer project to the base repository's topic-recognition branch. 

**👩‍💻 Reviewer's Feedback:**

I would express my gratitude to all the staff involved in the course this semester for their support and guidance. I have tried to do my best in the task and I would love to receive feedback to further improve and build upon my project! Thank you so much!
Had a great time working on this!

**🤔 Details:**

Name: Tarushi Gera
Student ID: 48242204

✅ **Checklist:**

- [x] modules.py - Contains the code for the classes of the Swin Architecture Model.
- [x] dataset.py - Contains the data loader for preprocessing the ADNC dataset.
- [x] train.py - Contains the code for training and validating the model on the dataset.
- [x] predict.py - Contains the code for example usage.
- [x] README.md - Documentation on the project.

🙌 **Thank you for your consideration!**

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
- Need to explain how Swin Transformer works and how your algorithm works.

Commit messages: Excellent, detailed.

Code:
- Uses Swin transformer
- Well commented
- Good model design

Functionality/Performance:
- Test accuracy is 72.27%
- Preprocssing data correctly
- Performed data augmentation
- No evidence show patient level split.
- Well done

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, but needs patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation could be better -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
